bin/runtest.ml

```diff
@@ -0,0 +1,76 @@
+open Import
+
+(** TODO-Ali: update this information for new features of test *)
+let info =
+  let doc = "Run tests." in
+  let man =
+    [ `S "DESCRIPTION"
+    ; `P "Run the given tests. The [TEST] argument can be either:"
+    ; `I
+        ( "-"
+        , "A directory: If a directory is provided, dune will recursively run all tests \
+           within that directory." )
+    ; `I
+        ( "-"
+        , "A file name: If a specific file name is provided, dune will run the tests \
+           with that name. The file name can be specified with or without its extension \
+           or its directory structure. This only applies to cram tests and not other \
+           kinds of tests. Other tests can be supported by adding an alias and asking \
+           for that instead." )
+    ; `P
+        "If no [TEST] is provided, dune will run all tests in the current directory and \
+         its subdirectories."
+    ; `P "See EXAMPLES below for additional information on use cases."
+    ; `Blocks Common.help_secs
+    ; Common.examples
+        [ "Run all tests in a given directory", "dune runtest DIR"
+        ; "Run a specific cram test", "dune runtest mytest.t"
+        ; "Run a specific alias (note the missing @ syntax):", "dune runtest myalias"
+        ; ( "Run all tests in the current source tree (including those that passed on \
+             the last run)"
+          , "dune runtest --force" )
+        ; ( "Run tests sequentially without output buffering"
+          , "dune runtest --no-buffer -j 1" )
+        ]
+    ]
+  in
+  Cmd.info "runtest" ~doc ~man ~envs:Common.envs
+;;
+
+let term =
+  let name = Arg.info [] ~docv:"TEST" in
+  let+ builder = Common.Builder.term
+  and+ dirs = Arg.(value & pos_all string [ "." ] name) in
+  let common, config = Common.init builder in
+  let request (setup : Import.Main.build_system) =
+    Action_builder.all_unit
+      (List.map dirs ~f:(fun dir ->
+         let dir = Path.(relative root) (Common.prefix_target common dir) in
+         Alias.request
+         @@
+         let base_dir, ext = Path.split_extension dir in
+         let alias_kind =
+           match ext, Path.is_directory dir with
```

Comment:
Instead of hard coding the extension, how about you use the source tree in conjunction with `Cram_rules.cram_tests`? That should be a bit more reliable.

Reply:
I've done what you've suggested instead.